receiver/k8sclusterreceiver/config.go

```diff
@@ -39,6 +39,11 @@ type Config struct {
 
 	// MetricsBuilderConfig allows customizing scraped metrics/attributes representation.
 	metadata.MetricsBuilderConfig `mapstructure:",squash"`
+
+	// Namespace to fetch resources from. If this is set, certain cluster-wide resources such as Nodes or Namespaces
+	// will not be able to be observed. Setting this option is recommended for sidecar deployment patterns where the collector runs
+	// as a sidecar for a pod, or in environments where due to security restrictions the collector can not be granted cluster-wide permissions.
```

Comment:
I don't like recommending the receiver be used in a sidecar.  Even if it is only scoped to getting namespace-specific metrics, if it is run in multiple sidecars within that namespace you'll produce duplicate metrics.

Reply:
Good point, I have removed that part from the comment